pkg/handler/project/project.go

```diff
@@ -1391,44 +1391,107 @@ func (h *handler) CreateWorkUnit(c *gin.Context) {
+		l.Error(ErrProjectNotFound, "project not found")
+		c.JSON(http.StatusNotFound, view.CreateResponse[any](nil, nil, ErrProjectNotFound, nil, ""))
+		return
+	}
+
+	tx, done := h.repo.NewTransaction()
+
+	workUnit := &model.WorkUnit{
+		Name:      input.Body.Name,
+		Type:      model.WorkUnitType(input.Body.Type),
+		Status:    model.WorkUnitStatus(input.Body.Status),
+		SourceURL: input.Body.URL,
+		ProjectID: model.MustGetUUIDFromString(input.ProjectID),
+	}
+
+	if err := h.store.WorkUnit.Create(tx.DB(), workUnit); err != nil {
+		l.Error(err, "failed to create new work unit")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, done(err), input, ""))
+		return
+	}
+
+	// create work unit stack
+	for _, stackID := range input.Body.Stacks {
+		stack, err := h.store.Stack.One(tx.DB(), stackID)
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			l.Error(err, "stack not found")
+			c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, done(ErrStackNotFound), input, ""))
+			return
+		}
+		if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
+			l.Error(err, "failed to get stacks by ID")
+			c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, done(err), input, ""))
+			return
+		}
+
+		wuStack := model.WorkUnitStack{
+			StackID:    model.MustGetUUIDFromString(stackID),
+			WorkUnitID: workUnit.ID,
+		}
+		if err := h.store.WorkUnitStack.Create(tx.DB(), &wuStack); err != nil {
+			l.Error(err, "failed to create new work unit stack")
+			c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, done(err), input, ""))
+			return
+		}
+
+		wuStack.Stack = *stack
+		workUnit.WorkUnitStacks = append(workUnit.WorkUnitStacks, wuStack)
+	}
+
+	// create work unit member
+	for _, employeeID := range input.Body.Members {
+		employee, err := h.store.Employee.One(tx.DB(), employeeID)
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			l.Error(err, "stack not found")
+			c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, done(ErrEmployeeNotFound), input, ""))
+			return
+		}
+		if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
+			l.Error(err, "failed to get employee by ID")
+			c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, done(err), input, ""))
+			return
+		}
+
+		wuMember := model.WorkUnitMember{
+			Status:     model.ProjectMemberStatusActive.String(),
+			WorkUnitID: workUnit.ID,
+			EmployeeID: model.MustGetUUIDFromString(employeeID),
+			ProjectID:  model.MustGetUUIDFromString(input.ProjectID),
+		}
+		if err := h.store.WorkUnitMember.Create(tx.DB(), &wuMember); err != nil {
+			l.Error(err, "failed to create new work unit stack")
+			c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, done(err), input, ""))
+			return
+		}
+
+		wuMember.Employee = *employee
+		workUnit.WorkUnitMembers = append(workUnit.WorkUnitMembers, wuMember)
+
+		// create project member
+		projectMember := model.ProjectMember{
```

Comment:
the list of employee ID we get from project slot(api get list member), so I think we don't need to create new project member. If the employee is not exist in project member just throw error. Help me to comfirm pls @namnhce